python/tvm/relax/frontend/onnx_frontend.py

```diff
@@ -0,0 +1,1150 @@
+        if not pad_mode in ["constant", "edge", "reflect"]:
+            raise tvm.error.OpAttributeInvalid(
+                "Value " + pad_mode + ' in attribute "mode" is invalid for operator Pad.'
+            )
+
+        if pad_mode == "constant":
+            return bb.emit_te(topi.nn.pad, inputs[0], pad_before, pad_after, constant_value)
+        elif pad_mode == "reflect":
+            return bb.emit_te(topi.nn.mirror_pad, inputs[0], pad_before, pad_after, "REFLECT")
+        else:
+            # TODO(gigiblender) Support edge mode.
+            raise NotImplementedError("Pad mode {} not implemented".format(pad_mode))
+
+class Tile(OnnxOpConverter):
+    """Converts an onnx Tile node into an equivalent Relax expression."""
+
+    @classmethod
+    def _impl_v13(cls, bb, inputs, attr):
+        reps = inputs[1]
+        if isinstance(reps, relax.Constant):
+            reps = reps.data.numpy().tolist()
+        else:
+            raise ValueError("Dynamic reps for Tile are supported yet.")
+        return bb.emit_te(topi.tile, inputs[0], reps)
+
+
+def _get_convert_map(opset):
+    return {
+        "MatMul": relay.frontend.onnx.MatMul,
+        "Concat": Concat,
+        "Add": Add,
+        "Mul": Mul,
+        "Cast": Cast,
+        "Gather": Gather,
+        "Gemm": Gemm,
+        "Reshape": relay.frontend.onnx.Reshape,
+        "Div": Div,
+        "Sigmoid": Sigmoid,
+        "Softmax": Softmax,
+        "Transpose": Transpose,
+        "Unsqueeze": Unsqueeze,
+        "Gelu": Gelu,
+        "BiasGelu": BiasGelu,
+        "Where": Where,
+        "Clip": Clip,
+        "Equal": Equal,
+        "Shape": Shape,
+        "Not": Not,
+        "Tanh": Tanh,
+        "Sqrt": Sqrt,
+        "Relu": Relu,
+        "Conv": Conv,
+        "Pow": Pow,
+        "Erf": Erf,
+        "CumSum": CumSum,
+        "Squeeze": Squeeze,
+        "Constant": Constant,
+        "Sub": Sub,
+        "LayerNormalization": relay.frontend.onnx.LayerNormalization,
+        "SkipLayerNormalization": relay.frontend.onnx.SkipLayerNormalization,
+        "EmbedLayerNormalization": relay.frontend.onnx.EmbedLayerNormalization,
+        # defs/reduction
+        "ReduceMax": relay.frontend.onnx.ReduceMax,
+        "ReduceMin": relay.frontend.onnx.ReduceMin,
+        "ReduceSum": relay.frontend.onnx.ReduceSum,
+        "ReduceMean": relay.frontend.onnx.ReduceMean,
+        "ReduceProd": relay.frontend.onnx.ReduceProd,
+        "ReduceLogSumExp": relay.frontend.onnx.ReduceLogSumExp,
+        "ReduceLogSum": relay.frontend.onnx.ReduceLogSum,
+        "ReduceSumSquare": relay.frontend.onnx.ReduceSumSquare,
+        "ReduceL1": relay.frontend.onnx.ReduceL1,
+        "ReduceL2": relay.frontend.onnx.ReduceL2,
+        "Expand": relay.frontend.onnx.Expand,
+        "ConstantOfShape": ConstantOfShape,
+        "Slice": Slice,
+        "Attention": relay.frontend.onnx.Attention,
+        "Pad": Pad,
+        "Split": Split,
+        "Tile": Tile,
+    }
+
+
+class GraphProto:
```

Comment:
i think this is almost the top-level converter glue class, however the name GraphProto shadows the pb2.GraphProto name. possible to pick a name that more accurately describes it?

Reply:
Renamed